test/test_filesystem_helper.cpp

```diff
@@ -495,15 +505,14 @@ TEST(TestFilesystemHelper, create_temp_directory)
   // edge case inputs
   {
     // Provided no base name we should still get a path with the 6 unique template chars
-    const auto tmpdir_emptybase = rcpputils::fs::create_temp_directory("");
+    const auto tmpdir_emptybase = rcpputils::fs::create_temporary_directory("");
     EXPECT_EQ(tmpdir_emptybase.filename().string().size(), 6u);
 
-    // Empty path doesn't exist and cannot be created
-    EXPECT_THROW(rcpputils::fs::create_temp_directory("basename", path()), std::system_error);
-
```

Comment:
Note: The empty path is a valid case for the new implementation. The temp folder will be created in the current folder. 
However, that is not good for tests when creating a new temporary folder from where the tests running. It could be a read-only volume or require special privileges for write access.